transport.go

```diff
@@ -102,6 +103,13 @@ func (t *Transport) challenge(req *http.Request) (*Challenge, int, bool) {
 	return cc.chal, cc.count, true
 }
 
+func (t *Transport) forgetChallenge(req *http.Request) {
+	t.cacheMu.Lock()
+	defer t.cacheMu.Unlock()
+	host := req.URL.Hostname()
+	delete(t.cache, host)
+}
```

Comment:
Instead of acquiring the lock again, let's just do this inside the `save` method. Also, let's have it forget the old challenge on any error. If `save` is being executed, then the previous challenge has already failed.